### PR TITLE
Add stack-trace formatter

### PR DESCRIPTION
**Pull request by aymanelamyaghri**

### CODE REVIEW

File: modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/RuntimeStepFilter.scala [hunk 1]
```diff
@@ -3,11 +3,13 @@ package ch.epfl.scala.debugadapter.internal.stepfilter
 import com.sun.jdi.Method
 import com.sun.jdi.Location
 import ch.epfl.scala.debugadapter.ScalaVersion
+import java.util.Optional
```
Comment (adpi2) [inline modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/RuntimeStepFilter.scala:6]:
```suggestion
```

File: modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/Scala3StepFilter.scala [hunk 2]
```diff
@@ -12,18 +12,33 @@ import scala.util.Try
...
       case e: InvocationTargetException => throw e.getCause
     }
+  }
+
+  override def formatScala(method: jdi.Method): Option[String] = {
+    try {
+
+      val optionalFormatting = formatMethod.invoke(bridge, method).asInstanceOf[Optional[String]]
+      if (!formatMethod.invoke(bridge, method).asInstanceOf[Optional[String]].isPresent()) None
+      else Some(optionalFormatting.get())
```
Comment (adpi2) [inline modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/Scala3StepFilter.scala:36]: If you import `scala.jdk.converters.OptionConverters._` you can then do

```suggestion
      formatMethod.invoke(bridge, method).toScala
```

File: modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/ScalaStepFilter.scala [hunk 3]
```diff
@@ -11,11 +11,39 @@ import com.sun.jdi.Method
...
+    else formatScala(method)
+  }
+  def formatScala(method: Method): Option[String] = Some(formatJava(method))
+  def formatJava(method: Method): String = {
+    val declaringType = method.declaringType().name.split("\\.").last
+    val methodName = method.name()
+    val argumentTypes = method.argumentTypes.asScala.toList
+      .map(t => t.name().split("\\.").last)
+      .mkString(",")
+    s"$declaringType.$methodName(${if (argumentTypes.nonEmpty) argumentTypes else ""})"
```
Comment (adpi2) [inline modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/ScalaStepFilter.scala:42]: Maybe add the return type to be consistent with Scala methods:

```suggestion
    s"$declaringType.$methodName(${if (argumentTypes.nonEmpty) argumentTypes else ""}): $returnType"
```

File: modules/scala-3-step-filter/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/ScalaStepFilterBridge.scala [hunk 4]
```diff
@@ -25,10 +28,117 @@ class ScalaStepFilterBridge(
 
   private def warn(msg: String): Unit = warnLogger.accept(msg)
 
-  private def throwOrWarn(msg: String): Unit =
-    if (testMode) throw new Exception(msg)
-    else warn(msg)
+  private def throwOrWarn(msg: String): Unit = {}
+//    if (testMode) throw new Exception(msg)
+  //  else warn(msg)
```
Comment (adpi2) [inline modules/scala-3-step-filter/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/ScalaStepFilterBridge.scala:33]: You should revert this change.

**Review by adpi2 — COMMENTED** [review]
That's probably the last batch of comments before we can merge this PR.

**Comment by adpi2** [comment]
Also you should run `scalafmt` before pushing to fix the CI.

File: modules/scala-3-step-filter/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/ScalaStepFilterBridge.scala [hunk 5]
```diff
@@ -25,10 +28,117 @@ class ScalaStepFilterBridge(
...
+    val method = jdi.Method(obj)
+    findSymbol(obj).map { t =>
+      val notMethodType = t.declaredType match {
+        case _: MethodType => false
+        case _: PolyType => false
+        case _ => true
+      }
+      val optionalString = if (notMethodType) ": " else ""
+
+      s"${t.owner.name}.${t.name}${optionalString}${formatType(t.declaredType)}"
```
[the remark this answers is not in the document]
Comment (aymanelamyaghri) [reply modules/scala-3-step-filter/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/ScalaStepFilterBridge.scala:139]: I tried this new formatting, but  it doesnt compile since t.owner doesnt have  a declareType  , how can we find the type?

**Review by adpi2 — APPROVED** [review]
